tpot2/tpot_estimator/estimator.py

```diff
@@ -476,6 +499,20 @@ def __init__(self,  scorers,
         self.callback = callback
         self.processes = processes
 
+        # create random number generator based on rng_seed
+        self.rng = np.random.default_rng(rng_seed)
+        self.rng_seed = rng_seed
+
+        # set the numpy seed so anything using it will be consistent as well
+        # not sure if this is 100% true tho
+        np.random.seed(rng_seed)
+
+        # lock in a random seed if cv_seed is None
+        if cv_seed is None:
+            self.cv_seed = self.rng.integers(1000000)
```

Comment:
This breaks sklearn conventions for estimators.

all hyper parameters must have a corresponding attribute that matches and is not changed. So in this case, self.cv_seed must equal cv_seed and should not be overwritten with the random value.

more information here: https://scikit-learn.org/stable/developers/develop.html

Reply:
Agreed.